### PR TITLE
Hide remember me

### PR DESCRIPTION
**Pull request by owencraston**
**Description**
- Remove remember me as an option for new users
- The key here is `new users`. Users who have already enabled remember me should not be affected by this change.
- This simply hides the front end that allows for remember me to be toggled on.
- Based on the feedback below, remember me will be something the user can turn on in settings and is now intended more for developers
- I also made a `LoginOptionsSwitch` that handles if what login option should be rendered into its own component since it was repeated in several files
- biometrics is always going to be the higher priority option for login. we will only show remember me as an option if its enabled in settings and biometrics are disabled

**Future improvements**
- Once the [auth refactor ships](https://github.com/MetaMask/metamask-mobile/pull/3755), we can move ALL of the biometric logic (state on weather to show biometric or not, handle errors etch) into this component. 
- The main blocker from doing this now is this [handleRejectedOsBiometricPrompt](https://github.com/MetaMask/metamask-mobile/blob/main/app/components/Views/ChoosePassword/index.js#L457) which calls [updateBiometryChoice](https://github.com/MetaMask/metamask-mobile/blob/main/app/components/Views/ChoosePassword/index.js#L557)
- the updateBiometryChoice function is repeated in each of these files

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**https://github.com/MetaMask/metamask-mobile/issues/4194**
Progresses https://github.com/MetaMask/metamask-mobile/issues/4194

### CODE REVIEW

**Review by blackdevelopa — COMMENTED** [review]
I didn't run this code and my comments are simply nitpicking.

File: app/components/UI/LoginWithBiometricsSwitch/styles.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable import/prefer-default-export */
+import { fontStyles } from '../../../styles/common';
+import { StyleSheet } from 'react-native';
+
+export const createStyles = (colors: any) =>
```
[the remark this answers is not in the document]
Comment (owencraston) [reply app/components/UI/LoginWithBiometricsSwitch/styles.ts:5]: I am pretty sure this is the new pattern. @Cal-L gave a knowledge share yesterday on the new component library and part of this is having a separate file for all the "parts" of the component. [Here](https://github.com/MetaMask/metamask-mobile/tree/main/app/component-library/components/BaseText) is a reference

File: app/components/Views/ChoosePassword/index.js [hunk 2]
```diff
@@ -564,52 +563,13 @@ class ChoosePassword extends PureComponent {
...
-              thumbColor={importedColors.white}
-              ios_backgroundColor={colors.border.muted}
-              testID={'remember-me-toggle'}
-            />
-          </>
-        )}
-      </View>
-    );
+    const { biometryType } = this.state;
+    return biometryType ? (
```
[the remark this answers is not in the document]
Comment (owencraston) [reply app/components/Views/ChoosePassword/index.js:567]: In general I prefer to use the ternary since using the logical operators (&& etc) can have weird side effects with falsely values. [Here is an article explaining the problem and solution](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx). The other benefit of this is it is a little more explicit in terms of readability and encourages componentization.

**Comment by gantunesr** [comment]
Some general notes on the PR,

- If this has an impact in the UI/UX, you should add a screenshot/recording of the before and after for QA and future references
- Don't use the keyword `Resolves`, GH closes the issue related to the PR when it is merged, and this can lead to some confusion on our product boards. Use `Progresses` instead.

Regarding the code, it looks good to me in the scope of my knowledge about the "Remember Me" epic. I'm not sure if you want @sethkfman to review it since he has more context on the Auth Refactor. If not, and @blackdevelopa agrees, we can approve it and pass it to QA after the failing test (`LoginWithBiometricsSwitch`) is solved.

**Review by gantunesr — DISMISSED** [review]
LGTM. @blackdevelopa let me know if you have any objection.

@owencraston will solve the failing test before moving it to QA

**Comment by AlexJupiter** [comment]
Suggested route forward to still allow this feature to be used by developers. 

1. Still hide this feature for new users, but provide the ability for someone to turn Remember Me on via a toggle on the Security & Privacy settings section of the app.
2. This toggle should be placed underneath the "Unlock with Face ID?" toggle and be named "Unlock with Remember Me?". It should also have the subtitle "Turn on at your own risk, since this feature gives anyone with access to your phone also access to MetaMask without any additional security."
3. This toggle should be turned off by default on new installations

@jakehaugen could you briefly check my copy suggestion please? Thank you :).

cc. @plasmacorral @gantunesr @sethkfman @owencraston

**Comment by AlexJupiter** [comment]
@owencraston slight improvement to the copy @jakehaugen and I have discussed:
Title: `Allow "Remember me"`
Subtitle: `Turn on at your own risk. When "Remember me" is on, it gives anyone access to your MetaMask if they can access your phone.`

Since this should just enable the checkbox again, and not actually turn it on, we think this copy improvement is necessary. Any thought let us know!

File: app/components/UI/LoginOptionsSwitch/LoginOptionsSwitch.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            true: colors.primary.default,
+            false: colors.border.muted,
+          }}
+          thumbColor={colors.white}
+          ios_backgroundColor={colors.border.muted}
+        />
+      </View>
+    );
+  }
+  return null;
```
Comment (owencraston) [inline app/components/UI/LoginOptionsSwitch/LoginOptionsSwitch.tsx:94]: This component does possibly return null. I felt it was valuable to make this API simple and handle all the display logic inside this component. If not we would have to have a similar if else sequence in each parent component.

File: app/components/Views/Settings/SecuritySettings/index.js [hunk 4]
```diff
@@ -821,6 +835,15 @@ class Settings extends PureComponent {
     );
   };
 
+  renderRememberMeOptionSection = () => (
+    <SecurityOptionToggle
```
Comment (owencraston) [inline app/components/Views/Settings/SecuritySettings/index.js:839]: I built this component with the goal of replacing all the duplicated view logic inside this file but felt that I wanted to keep the changes small for the sake of velocity. Let me know if I should swap them out in this PR

**Review by gantunesr — COMMENTED** [review]
LGTM. Left small nits

I notice you're introducing new patterns and standards in your code that can have a positive overall impact, would you care to draft a proposal and present it to the team so we can all follow the same coding practices.

File: app/components/UI/LoginOptionsSwitch/LoginOptionsSwitch.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            false: colors.border.muted,
+          }}
+          thumbColor={colors.white}
+          ios_backgroundColor={colors.border.muted}
+        />
+      </View>
+    );
+  } else if (
+    shouldRenderBiometricOption === undefined &&
+    allowLoginWithRememberMe === true
```
[the remark this answers is not in the document]
Comment (owencraston) [reply app/components/UI/LoginOptionsSwitch/LoginOptionsSwitch.tsx:73]: Screen Shot 2022-06-13 at 11 44 30 AM
The only available values for this variable is for it to be defined as one of the `BIOMETRY_TYPE` or to be undefined, therefore I think its safe to say this will catch all the cases.

> Why is it necessary to compare to true? Would ... && allowLoginWithRememberMe be enough?
You are totally correct. I felt that the verbosity made it a bit easier to understand the logic. Up to you if you think it adds value.

File: app/components/Views/Settings/SecuritySettings/index.js (hunk 4 above)
Comment (sethkfman) [reply app/components/Views/Settings/SecuritySettings/index.js:839]: I think that is a good strategy.

**Comment by plasmacorral** [comment]
### Observation M1

Getting an error when attempting to change the password:
"View:Root
TypeError: null is not an object (evaluating 'shouldRenderBiometricOption.toLowerCase')"

https://recordit.co/5SPXcE2iIL

Reproduction steps:
1. Configure account with v5.2 build 913, import an account, bind Keystone hardware wallet and enable remember me
2. Update to the branch under test
3. launch MetaMask mobile
4. tap the burger menu
5. tap settings
6. tap security and privacy
7. tap change password
8. observe error

**Comment by owencraston** [comment]
@plasmacorral With my latest commit, I have addressed observation M1.

File: app/util/password/index.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+export const passwordRequirementsMet = (password: string) =>
+  password.length >= MIN_PASSWORD_LENGTH;
+
+interface PasswordValidationResponse {
+  valid: boolean;
+  message: string;
+}
+
+export const doesPasswordMatch = async (
```
Comment (owencraston) [inline app/util/password/index.ts:30]: I debated turning this into a hook so we could memoize some of the values from the SecureKeychain and Engine but it feels more like a util function. Making it a util will also make it easier to test.

File: app/components/UI/LoginOptionsSwitch/LoginOptionsSwitch.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            false: colors.border.muted,
+          }}
+          thumbColor={colors.white}
+          ios_backgroundColor={colors.border.muted}
+        />
+      </View>
+    );
+  } else if (
+    shouldRenderBiometricOption === null &&
+    allowLoginWithRememberMe === true
```
Comment (gantunesr) [inline app/components/UI/LoginOptionsSwitch/LoginOptionsSwitch.tsx:73]: is `allowLoginWithRememberMe` already a boolean? The comparison seems to be unnecessary

File: app/components/UI/TurnOffRememberMeModal/TurnOffRememberMeModal.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+  const modalRef = useRef<ReusableModalRef>(null);
+
+  const [passwordText, setPasswordText] = useState<string>('');
+  const [disableButton, setDisableButton] = useState<boolean>(true);
+
+  const isValidPassword = useCallback(async (text: string) => {
+    const response = await doesPasswordMatch(text);
+    return response.valid;
+  }, []);
```
Comment (gantunesr) [inline app/components/UI/TurnOffRememberMeModal/TurnOffRememberMeModal.tsx:36]: Add return type (`Promise<boolean>`)

File: ios/Podfile.lock [hunk 9]
```diff
@@ -841,4 +841,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 5759da19ce141e5f8aca3a4f56ac232baa34e957
 
-COCOAPODS: 1.11.3
+COCOAPODS: 1.11.2
```
Comment (gantunesr) [inline ios/Podfile.lock:844]: Revert

File: app/components/Views/Login/checkIfUsingRememberMe.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    // Restore vault with existing credentials
+    const { KeyringController } = Engine.context as any;
+    try {
+      await KeyringController.submitPassword(credentials.password);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  } else return false;
+};
```
Comment (gantunesr) [inline app/components/Views/Login/checkIfUsingRememberMe.ts:20]: 1. Is this a utils method?
2. Is there any value in logging the error?
3. Add return type

**Comment by gantunesr** [comment]
Some issues I noticed while testing the branch,

1. The "Biometrics" is getting triggered when I try to access the Security & Privacy settings
2. When I enable "Unlock with Face ID" it triggers the Biometrics two times
3. The password modal gets out of the screen (See image for reference)
4. Pressing outside of the password modal throws an error. The error is `TypeError: trigger.current is not a function. (In 'trigger.current()', 'trigger.current' is an instance of Class)`. I see that this is a reference error regarding the `ReusableModal`.

cc @plasmacorral

**Review by gantunesr — COMMENTED** [review]
Good work mate, I left a initial review for now, there are some files I want to take a deeper look tomorrow

File: app/components/Views/Login/checkIfUsingRememberMe.ts (hunk 10 above)
Comment (owencraston) [reply app/components/Views/Login/checkIfUsingRememberMe.ts]: I feel that this is a util function but I am not sure where it should live. One could argue that it could replace the similarly named method in side the Login/index file but I don't want to touch such sensitive code in the pursuit of shipping quickly. What do you think @sethkfman ?

File: app/components/Views/Login/checkIfUsingRememberMe.ts [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+import AsyncStorage from '@react-native-community/async-storage';
+import { BIOMETRY_CHOICE } from '../../../constants/storage';
+/**
+ * Checks to see if the user has enabled Remember Me and logs
+ * into the application if it is enabled.
+ */
+export const checkIfRememberMeEnabled = async () => {
+  const biometryChoice = await AsyncStorage.getItem(BIOMETRY_CHOICE);
+  // since we do not allow remember me to be used with biometrics we can eagerly return false
+  if (biometryChoice) return false;
```
Comment (owencraston) [inline app/components/Views/Login/checkIfUsingRememberMe.ts:13]: @gantunesr this will prevent the biometrics from being called on the setting screen.

File: bitrise.yml [hunk 12]
```diff
@@ -323,10 +323,10 @@ app:
           PROJECT_LOCATION_IOS: ios
         - opts:
               is_expand: false
-          VERSION_NAME: 5.3.0
+          VERSION_NAME: 5.3.1
         - opts:
               is_expand: false
-          VERSION_NUMBER: 927
+          VERSION_NUMBER: 928
```
Comment (gantunesr) [inline bitrise.yml:329]: Leaving a comment to remember to revert this change before merging

**Review by gantunesr — COMMENTED** [review]
Just a small comment

File: app/components/UI/TurnOffRememberMeModal/TurnOffRememberMeModal.tsx [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+          <View style={styles.areYouSure}>
+            <Text style={[styles.heading, styles.delete]}>
+              {strings('turn_off_remember_me.description')}
+            </Text>
+            <OutlinedTextField
+              style={styles.input}
+              testID={'TurnOffRememberMeConfirm'}
+              autoFocus
+              returnKeyType={'done'}
+              onChangeText={checkPassword}
```
[the remark this answers is not in the document]
Comment (owencraston) [reply app/components/UI/TurnOffRememberMeModal/TurnOffRememberMeModal.tsx:93]: **it will at most execute every 2 milliseconds**

File: app/util/password/index.js [hunk 14]
```diff
@@ -1,18 +0,0 @@
...
-    case 2:
-      return 'weak';
-    case 3:
-      return 'good';
-    case 4:
-      return 'strong';
-  }
-};
-
-export const passwordRequirementsMet = (password) =>
```
Comment (owencraston) [inline app/util/password/index.js:17]: I just converted this file to ts which was trivial.

**Comment by owencraston** [comment]
@plasmacorral I am ready for you to take another look 👀 !

File: app/components/UI/TurnOffRememberMeModal/TurnOffRememberMeModal.tsx [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+        cancelText={strings('turn_off_remember_me.action')}
+        cancelButtonDisabled={disableButton}
+        onCancelPress={disableRememberMe}
+        onRequestClose={triggerClose}
+        onConfirmPress={triggerClose}
+      >
+        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+          <View style={styles.areYouSure}>
+            <Text style={[styles.heading, styles.delete]}>
+              {strings('turn_off_remember_me.description')}
```
[the remark this answers is not in the document]
Comment (owencraston) [reply app/components/UI/TurnOffRememberMeModal/TurnOffRememberMeModal.tsx:86]: I think the description would ring better if it said something like this...
Title: "Type your password to disable Remember Me"
Description: You will be locked out of MetaMask and will need your password to unlock MetaMask going forward.

**Review by gantunesr — COMMENTED** [review]
LGTM

File: bitrise.yml [hunk 16]
```diff
@@ -323,10 +323,10 @@ app:
           PROJECT_LOCATION_IOS: ios
         - opts:
               is_expand: false
-          VERSION_NAME: 5.3.0
```
Comment (owencraston) [inline bitrise.yml:326]: needs to be reverted before merge.

**Comment by plasmacorral** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by plasmacorral** [comment]
I have completed another pass of feature QA and have the following observations:

MB2: Password entry not obfuscated in the prompt after attempting to disable RM enable toggle.  Owen mentioned some remaining UI work, but this must be completed prior to merging for RC.


MB3: faceID/fingerprint authentication is turned on during a password change, regardless of user selection during the password change or configuration prior to password change
- faceID turned on in pw change causing device to freeze on orange diamond when faceID perms are not granted or have been previously disabled https://recordit.co/qDk2SKyemb
- faceID turned on in pw change when permissions have previously been granted https://recordit.co/z9mXoPCQ5a
- https://recordit.co/IPGozRiBYc faceID toggle within password change is:
              - 16s- Showing the toggle set to off when faceID auth is enabled
              - 45s- faceID remains enabled if user engages the toggle to on
              - 1m5s- Only turns off faceID if the user taps the toggle on then off.
- This also happens when Device passcode is the authentication factor
              - https://recordit.co/EcUzoY1h7C
              - 5s login and toggle RM feature to on
              - 20s enable iPhone passcode auth
              - 25s use iPhone passcode auth for access
              - 28s accessing Security & Privacy requires device passcode 
              - 35s confirm password for a pw change with Device passcode auth enabled
              - 42s enter and confirm new password, and do NOT enable "unlock with Face ID"

MB4: https://recordit.co/EcUzoY1h7C in the device passcode video at 28s, you can see an example of the device passcode being prompted to allow the user into the Security & Privacy menu. 
    - This is not the case today in v5.3 build 927, where a user with device passcode authentication is able to open the settings page without having to provide the device passcode. https://recordit.co/X23ksVCcnh

MB5: Revisiting a recording from MB3 https://recordit.co/IPGozRiBYc.   The Biometric/faceID toggle within the change password flow does not match current user settings 
    - If biometrics or device passcode are enabled in Production build 927 v5.3, this toggle defaults to ON within the password change flow and matches the current user settings.

MB6: on IOS cannot use password auth as an alternative to faceID until turning faceID on, then back off- bug?
    - https://recordit.co/qWOrEsplxt

MB7: When tapping "cancel" on faceID failure, toggle at login page looks like faceID is off...but faceID actually remains enabled.  
    - User must turn on faceID and then turn it off again to use password authentication. https://recordit.co/Y5FqxH9aWt

MB8:  Device passcode authentication enables RM enable toggle https://recordit.co/B5FscoUN5g  
    - @40s- Turning on device passcode auth turns the enable remember me toggle to ON.  Should these two setting toggles be independent?

MB9: from the same Device passcode authentication video https://recordit.co/B5FscoUN5g  
    - @41s-  When device passcode is auth method, passcode is asked after every keystroke when trying to disable RM.  

MB10: When biometrics are present the faceID toggle in the import from seed view should default to the ON position.
    - It does default to on in production build 927, but not in this branch.

<div style="display: inline-block">

**Comment by owencraston** [comment]
@holantonela I roughly implemented your design requests. Does this look good to you?
Image from iOS

**Comment by holantonela** [comment]
Thanks for this iteration @owencraston. it looks LGTM! 

We will need a password default text field. We can do it now if a reusable component exist or open a ticket.

**Comment by holantonela** [comment]
Ay, Corey revisited the copy. We should have 


`After it, MetaMask must lock. You will use your password to unlock MetaMask going forward.`

**Comment by plasmacorral** [comment]
MB 10:
Just noticed in the screenshots from observation MB9 that the actual toggle is moved from the right side of the screen and in this branch the toggle appears underneath "Unlock with FaceID" on the left side.

Should the toggle remain on the left side of the screen?

**Comment by georgewrmarshall** [comment]
> Thanks for this iteration @owencraston. it looks LGTM!
> 
> We will need a password default text for the input field. We can do it now if a reusable component exists or open a ticket for the design system team to make one. cc @georgewrmarshall

@holantonela I think @owencraston just needs to add a `placeholder` prop to `OutlinedTextField` component? Not super familiar with that component but looks like it accepts it?

**Comment by owencraston** [comment]
@plasmacorral regarding MB6:
This is the 5.4.0 build and it appears to be doing the same behaviour. Do you spot any differences?

https://user-images.githubusercontent.com/22918444/181635230-16a3063a-695c-46d7-8766-edbd8a9ce09e.MP4

**Comment by plasmacorral** [comment]
> @plasmacorral regarding MB6: This is the 5.4.0 build and it appears to be doing the same behaviour. Do you spot any differences?
> Image.from.iOS.MP4

Your video looks like it addresses MB6.

**Comment by owencraston** [comment]
@holantonela @georgewrmarshall 

Updated designs with new copy and placeholder

**Comment by owencraston** [comment]
I made a separate PR for the strings so we could get the translations in ASAP. I will rebase once this is [merged](https://github.com/MetaMask/metamask-mobile/pull/4784)

File: locales/languages/en.json [hunk 17]
```diff
@@ -2143,5 +2147,11 @@
   "download_files": {
     "error": "Failed to download attachment.",
     "unknownError": "Unknown error occurred"
+  },
```
Comment (owencraston) [inline locales/languages/en.json:2150]: these will be removed in favour of the strings from [this PR](https://github.com/MetaMask/metamask-mobile/pull/4784)

File: app/components/Views/ImportFromSeed/index.js [hunk 18]
```diff
@@ -394,10 +394,6 @@ class ImportFromSeed extends PureComponent {
     }
   };
 
-  onBiometryChoiceChange = (value) => {
```
Comment (owencraston) [inline app/components/Views/ImportFromSeed/index.js:397]: this method was completely unused. I imagine [updateBiometryChoice](https://github.com/MetaMask/metamask-mobile/blob/main/app/components/Views/ImportFromSeed/index.js#L448) that it was left when we migrated to this method.

(here is a search for the method on main and as you can see its only defined)
Screen Shot 2022-08-02 at 2 40 56 PM

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by owencraston** [comment]
Waiting for the translations to be merged in [this PR](https://github.com/MetaMask/metamask-mobile/pull/4262)

File: android/app/build.gradle [hunk 19]
```diff
@@ -160,8 +160,8 @@ android {
 		applicationId "io.metamask"
 		minSdkVersion rootProject.ext.minSdkVersion
 		targetSdkVersion rootProject.ext.targetSdkVersion
-		versionCode 939
-		versionName "5.5.0"
+		versionCode 942
```
Comment (owencraston) [inline android/app/build.gradle:163]: Must be removed before merging
